beat-that-script.js

```diff
@@ -0,0 +1,149 @@
+var compareScoresState = "compareScoresState";
+var gameState = rollDiceState;
+
+var currentPlayerRolls = [];
+var currentPlayer = 1;
+var bothPlayersScore = [];
+
+// reset game automatically
+var resetGame = function () {
+  currentPlayer = 1;
+  gameState = rollDiceState;
+  bothPlayersScore = [];
+};
+
+//Dice roll function
+var rollDice = function () {
+  var randomDecimal = Math.random() * 6;
+  var randomInteger = Math.floor(randomDecimal) + 1;
+
+  return randomInteger;
+};
+
+//Helper function to store both dice numbers in an array
+var rollBothDice = function () {
+  var counter = 0;
+  while (counter < 2) {
+    currentPlayerRolls.push(rollDice());
+    counter = counter + 1;
+  }
+  return (
+    `Welcome, Player ` +
+    currentPlayer +
+    `.` +
+    `<br><br>You rolled: ${currentPlayerRolls[0]} for Dice One and ${currentPlayerRolls[1]} for Dice Two. <br><br> Next, please input '1' if you would like to choose Dice One and '2' if you would like to choose Dice Two as the first digit of your final value.`
+  );
+};
+
+//Helper function to get player's score
+var getPlayerScore = function (playerInput) {
+  var playerScore;
+  //input validation
+  if (playerInput != 1 && playerInput != 2) {
+    console.log(`Control flow: input !== 1 && input !== 2`);
+    return `Error! Please input:<br>'1' if you want to use Dice One as the first digit of your final value or<br> '2' if you want to use Dice Two as the first digit of your final value.<br><br>You rolled: <br>${currentPlayerRolls[0]} for Dice One and <br>
+    ${currentPlayerRolls[1]} for Dice Two.`;
+  }
+
+  //input == 1
+  if (playerInput == "1") {
+    playerScore = Number(
+      String(currentPlayerRolls[0]) + String(currentPlayerRolls[1])
+    );
+  }
+
+  //input == 2
+  if (playerInput == "2") {
+    playerScore = Number(
+      String(currentPlayerRolls[1]) + String(currentPlayerRolls[0])
+    );
+  }
+
+  //Store player's score in array
+  bothPlayersScore.push(playerScore);
+  //clear current player's rolls array
+  currentPlayerRolls = [];
+  return (
+    `Player ` + currentPlayer + ` Your chosen value is: ` + playerScore + `.`
+  );
+};
+
+//Helper function to compare players' scores
+var comparePlayersScores = function () {
+  var compareMessage =
+    `Player 1's score: ` +
+    bothPlayersScore[0] +
+    `<br>Player 2's score: ` +
+    bothPlayersScore[1];
+
+  //Player 1 wins
+  if (bothPlayersScore[0] > bothPlayersScore[1]) {
+    compareMessage = compareMessage + `<br><br>Player 1 wins!`;
+  }
+
+  //Player 2 wins
+  if (bothPlayersScore[1] > bothPlayersScore[0]) {
+    compareMessage = compareMessage + `<br><br>Player 2 wins!`;
+  }
+
+  //Player 1 and 2 draw
+  if (bothPlayersScore[1] == bothPlayersScore[0]) {
+    compareMessage = compareMessage + `<br><br>It's a tie.`;
+  }
+  return compareMessage;
+};
+
+var main = function (input) {
+  console.log(`Checking gameState on submit`, gameState);
+  console.log(`Checking currentPlayer on submit`, currentPlayer);
```

Comment:
Good use of console.log to check on values. 

However, we generally remove console.log() before submitting assignments. Remember, they are like scaffolding when a building is being constructed which is removed once completed!